engine/hatchery/swarm/monitoring.go

```diff
@@ -0,0 +1,176 @@
+package swarm
+
+import (
+	"encoding/json"
+	"io"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/tag"
+	"golang.org/x/net/context"
+
+	"github.com/docker/docker/api/types"
+	"github.com/rockbears/log"
+
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/telemetry"
+)
+
+const (
+	TagNodeName     string = "node_name"
+	TagJobID        string = "job_id"
+	TagWorkerName   string = "worker_name"
+	TagResourceName string = "resource_name"
+)
+
+func (h *HatcherySwarm) InitWorkersMetrics(ctx context.Context) error {
+	h.workerMetrics.CPU = stats.Float64("cds/hatchery/worker_cpu", "number of cpu for a worker resource", stats.UnitDimensionless)
+	h.workerMetrics.CPURequest = stats.Float64("cds/hatchery/worker_cpu_request", "number of cpu requested for a worker resource", stats.UnitDimensionless)
+	h.workerMetrics.Memory = stats.Int64("cds/hatchery/worker_memory", "number of memory for a worker resource", stats.UnitDimensionless)
+	h.workerMetrics.MemoryRequest = stats.Int64("cds/hatchery/worker_memory_request", "number of memory requested for a worker resource", stats.UnitDimensionless)
+
+	tags := []tag.Key{
+		telemetry.MustNewKey(telemetry.TagServiceName),
+		telemetry.MustNewKey(telemetry.TagServiceType),
+		telemetry.MustNewKey(TagNodeName),
+		telemetry.MustNewKey(TagJobID),
+		telemetry.MustNewKey(TagWorkerName),
+		telemetry.MustNewKey(TagResourceName),
+	}
+
+	return telemetry.RegisterView(ctx,
+		telemetry.NewViewLastFloat64("cds/hatchery/worker_cpu", h.workerMetrics.CPU, tags),
+		telemetry.NewViewLastFloat64("cds/hatchery/worker_cpu_request", h.workerMetrics.CPURequest, tags),
+		telemetry.NewViewLast("cds/hatchery/worker_memory", h.workerMetrics.Memory, tags),
+		telemetry.NewViewLast("cds/hatchery/worker_memory_request", h.workerMetrics.MemoryRequest, tags),
+	)
+}
+
+func (h *HatcherySwarm) StartWorkerMetricsRoutine(ctx context.Context) {
+	ticker := time.NewTicker(5 * time.Second)
```

Comment:
could you put this duration in configuration file ?

Reply:
done